onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -41,10 +41,10 @@ ONNX_OPERATOR_KERNEL_EX(
 MKLDNNExecutionProvider::MKLDNNExecutionProvider(const MKLDNNExecutionProviderInfo& info)
     : IExecutionProvider{onnxruntime::kMklDnnExecutionProvider} {
   DeviceAllocatorRegistrationInfo default_allocator_info({OrtMemTypeDefault,
-                                                          [](int) { return std::make_unique<CPUAllocator>(std::make_unique<OrtAllocatorInfo>(MKLDNN, OrtAllocatorType::OrtDeviceAllocator, 0, OrtMemTypeDefault)); }, std::numeric_limits<size_t>::max()});
+                                                          [](int) { return std::make_unique<CPUAllocator>(std::make_unique<OrtAllocatorInfo>(MKLDNN, OrtAllocatorType::OrtDeviceAllocator)); }, std::numeric_limits<size_t>::max()});
 
   DeviceAllocatorRegistrationInfo cpu_allocator_info({OrtMemTypeCPUOutput,
-                                                      [](int) { return std::make_unique<CPUAllocator>(std::make_unique<OrtAllocatorInfo>(MKLDNN_CPU, OrtAllocatorType::OrtDeviceAllocator, 0, OrtMemTypeCPUOutput)); }, std::numeric_limits<size_t>::max()});
+                                                      [](int) { return std::make_unique<CPUAllocator>(std::make_unique<OrtAllocatorInfo>(MKLDNN_CPU, OrtAllocatorType::OrtDeviceAllocator, OrtDevice(), 0, OrtMemTypeCPUOutput)); }, std::numeric_limits<size_t>::max()});
 
```

Comment:
Why do we need default_allocator_info and cpu_allocator_info if they're both using the CPUAllocator? Given that, is there any difference between OrtMemTypeDefault and OrtMemTypeCPUOutput?

Reply:
this is exactly a serious problem that I'm solving. I'll come out of a doc to clarify this.

In short, it was because the previous design is kind of ruined, which leads to differences have to be represented for OrtMemTypeDefault and OrtMemTypeCPUOutput in allocator layer, which I believe is totally wrong......

Reply:
OrtMemTypeDefault is asking CPU memory, while OrtMemTypeCPUOutput is also asking CPU memory, but may be "Cuda_Pinned" for cuda execution provider. however, for nGraph/mkldnn, there's no such "Pinned" memory concept, I believe.

for this PR, I'll keep it as it is following the legacy logic and will do the refactoring in next PRs.